<?xml version="1.0"?>
<launch>

	<node pkg="cob_simulated_tactile_sensors" type="gazebo_virtual_tactile_sensors.py" name="tactile_sensors" ns="sdh_controller" output="screen">
		<!-- The number of patches on the tactile sensor in the direction perpendicular to the finger -->
		<param name="cells_x" value="6" />

		<!-- The number of patches on the tactile sensor along the direction of the finger -->
		<param name="cells_y" value="14" />
		
		<!-- The change of output in one patch per Newton -->
		<param name="sensitivity" value="350" />
		
		<!-- The maximum output value of one patch -->
		<param name="output_range" value="3500.0" />
		
		<!-- Length of the moving average filter which smoothes simulation output -->
		<param name="filter_length" value="10" />
		
		<!-- Remap the output topic -->
		<remap from="tactile_data" to="/sdh_controller/tactile_data" />
		
		<!-- Remap the bumper subscriptions to the simulation -->
		<remap from="finger_12/state" to="/sdh_finger_12_bumper/state" />
		<remap from="finger_13/state" to="/sdh_finger_13_bumper/state" />
		<remap from="thumb_2/state" to="/sdh_thumb_2_bumper/state" />
		<remap from="thumb_3/state" to="/sdh_thumb_3_bumper/state" />
		<remap from="finger_22/state" to="/sdh_finger_22_bumper/state" />
		<remap from="finger_23/state" to="/sdh_finger_23_bumper/state" />
	</node>

</launch>
